data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/handler/ContractAgreementHandler.java

```diff
@@ -101,28 +100,16 @@ public boolean canHandle(@NotNull MultipartRequest multipartRequest) {
         if (result.failed()) {
             monitor.debug(String.format("Could not transform contract agreement: [%s]",
                     String.join(", ", result.getFailureMessages())));
-            return createBadParametersErrorMultipartResponse(message);
+            return createMultipartResponse(badParameters(message, connectorId));
         }
 
         // TODO get hash from message
         var output = result.getContent();
         var processId = message.getTransferContract();
-        var negotiationResponse = negotiationManager.confirmed(claimToken,
+        var negotiationConfirmResult = negotiationManager.confirmed(claimToken,
```

Comment:
note for the future (after project structure review), here the `ContractNegotiationService` should be used instead of the manager